### PR TITLE
feat: migrate less to token for Skeleton

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- https://github.com/ant-design/ant-design/issues/41884

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | migrate less to token for Skeleton  |
| 🇨🇳 Chinese | 迁移 `Skeleton` 组件的 `less token`  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 379d44f</samp>

This pull request refactors the `Skeleton` component style to use the `ComponentToken` type and updates the documentation for migrating from Less variables to Component Token in both English and Chinese.

### <samp>🤖 Generated by Copilot at 379d44f</samp>

*  Add `skeletonTitleHeight` and `skeletonBlockRadius` properties to `ComponentToken` type to customize skeleton blocks ([link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-e9607dd76735d305ab98d05211d7ebaf400d805eb8c30e2e667b7b5f27814811R9-R10))
*  Remove redundant properties from `SkeletonToken` interface and adjust `token` object assignment accordingly ([link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-e9607dd76735d305ab98d05211d7ebaf400d805eb8c30e2e667b7b5f27814811L28-L29), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-e9607dd76735d305ab98d05211d7ebaf400d805eb8c30e2e667b7b5f27814811L356-L357), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-e9607dd76735d305ab98d05211d7ebaf400d805eb8c30e2e667b7b5f27814811R370-R371))
*  Add documentation for `Skeleton` component token mapping in English and Chinese versions of `docs/react/migrate-less-variables.md` ([link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL148-R167), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L148-R167))
*  Remove empty lines from various sections of `docs/react/migrate-less-variables.md` for both languages for formatting consistency ([link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL42), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL56), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL137), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL167), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L42), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L56), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L137), [link](https://github.com/ant-design/ant-design/pull/42134/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L167))

### CODE REVIEW

File: docs/react/migrate-less-variables.en-US.md [hunk 1]
```diff
@@ -145,15 +142,28 @@ This document contains the correspondence between all the less variables related
...
 
-<!-- ### Slider -->
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@skeleton-avatar-prefix-cls` | `skeletonAvatarCls` | - |
+| `@skeleton-paragraph-prefix-cls` | `skeletonParagraphCls` | - |
+| `@skeleton-button-prefix-cls` | `skeletonButtonCls` | - |
+| `@skeleton-input-prefix-cls` | `skeletonInputCls` | - |
+| `@skeleton-image-prefix-cls` | `skeletonImageCls` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:158]: 这些应该要去掉吧

Comment (kiner-tang) [reply docs/react/migrate-less-variables.en-US.md]: done

File: docs/react/migrate-less-variables.en-US.md [hunk 2]
```diff
@@ -145,15 +142,23 @@ This document contains the correspondence between all the less variables related
...
 
-<!-- ### Slider -->
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@skeleton-block-radius` | `skeletonBlockRadius` | - |
+| `@skeleton-title-height` | `skeletonTitleHeight` | - |
+| `@skeleton-color` | `color` | - |
+| `@skeleton-title-paragraph-margin-top` | `skeletonParagraphMarginTop` | - |
+| `@skeleton-paragraph-li-height` | `skeletonParagraphLineHeight` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:158]: 这两个没在 Component Token 哦

Comment (kiner-tang) [reply docs/react/migrate-less-variables.en-US.md]: done

File: docs/react/migrate-less-variables.zh-CN.md [hunk 3]
```diff
@@ -145,15 +142,21 @@ title: Less 变量迁移 Design Token
 | `@result-subtitle-font-size` | `resultSubtitleFontSize` | - |
 | `@result-extra-margin` | `resultExtraMargin` | - |
 
-
 <!-- ### Segment 分段器 -->
 
 <!-- ### Select 选择器 -->
 
-<!-- ### Skeleton 骨架屏 -->
+## Skeleton 骨架屏
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:149]: 这个好像看漏了，应该都是 ###

Comment (kiner-tang) [reply docs/react/migrate-less-variables.zh-CN.md:149]: 好，我都一起改一下

File: docs/react/migrate-less-variables.en-US.md [hunk 4]
```diff
@@ -140,7 +140,15 @@ This document contains the correspondence between all the less variables related
...
-<!-- ### Skeleton -->
+### Skeleton
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@skeleton-block-radius` | `blockRadius` | - |
+| `@skeleton-title-height` | `titleHeight` | - |
+| `@skeleton-color` | `color` | - |
+| `@skeleton-to-color` | `colorGradientEnd` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:151]: 这两个要改下命名了，保留现有的，打上 deprecated，新增 `gradientFromColor` `gradientToColor`

**Comment by MadCcc** [comment]
image
还有一些间距也可以迁过来

File: docs/react/migrate-less-variables.en-US.md (hunk 4 above)
Comment (kiner-tang) [reply docs/react/migrate-less-variables.en-US.md:151]: 好的

**Comment by kiner-tang** [comment]
> image 还有一些间距也可以迁过来

done

File: docs/react/migrate-less-variables.en-US.md [hunk 5]
```diff
@@ -154,7 +154,18 @@ This document contains the correspondence between all the less variables related
...
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@skeleton-block-radius` | `blockRadius` | - |
+| `@skeleton-title-height` | `titleHeight` | - |
+| `@skeleton-color` | `gradientFromColor` | - |
+| `@skeleton-to-color` | `gradientToColor` | - |
+| `@skeleton-paragraph-margin-top` | `paragraphMarginTop` | - |
+| `@skeleton-paragraph-li-height` | `paragraphLineHeight` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:167]:
```suggestion
| `@skeleton-paragraph-li-height` | `paragraphLiHeight` | - |
```

File: components/skeleton/demo/componentToken.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+const App: React.FC = () => (
+  <ConfigProvider
+    theme={{
+      components: {
+        Skeleton: {
+          blockRadius: 30,
+          titleHeight: 30,
+          color: '#222',
+          colorGradientEnd: '#444',
```
Comment (MadCcc) [inline components/skeleton/demo/componentToken.tsx:12]: 这里也改下哈

Comment (kiner-tang) [reply components/skeleton/demo/componentToken.tsx:12]: 好的，这个搞漏了

File: components/skeleton/style/index.tsx [hunk 7]
```diff
@@ -365,10 +365,15 @@ export default genComponentStyleHook(
   },
   (token) => {
     const { colorFillContent, colorFill } = token;
-
     return {
       color: colorFillContent,
       colorGradientEnd: colorFill,
+      gradientFromColor: colorFillContent,
+      gradientToColor: colorFill,
```
Comment (MadCcc) [inline components/skeleton/style/index.tsx:372]: 实际使用的地方都改成新的，废弃的两个在这里初始化之后赋值给新的 token，就可以同时生效了

File: components/theme/util/genComponentStyleHook.ts [hunk 8]
```diff
@@ -43,6 +44,8 @@ export default function genComponentStyleHook<ComponentName extends OverrideComp
     | ((token: GlobalToken) => OverrideTokenWithoutDerivative[ComponentName]),
   options?: {
     resetStyle?: boolean;
+    // Deprecated token key map {[oldTokenKey]: newTokenKey}
+    deprecatedTokens?: Record<string, string>;
```
Comment (MadCcc) [inline components/theme/util/genComponentStyleHook.ts:48]:
```suggestion
    deprecatedTokens?: [OverrideTokenWithoutDerivative[ComponentName], OverrideTokenWithoutDerivative[ComponentName]];
```
直接上元组吧

File: components/theme/util/genComponentStyleHook.ts [hunk 9]
```diff
@@ -74,6 +77,25 @@ export default function genComponentStyleHook<ComponentName extends OverrideComp
           typeof getDefaultToken === 'function' ? getDefaultToken(proxyToken) : getDefaultToken;
         const mergedComponentToken = { ...defaultComponentToken, ...token[component] };
 
+        if (options?.deprecatedTokens) {
+          const { deprecatedTokens } = options;
+          Object.keys(deprecatedTokens).forEach((oldTokenKey) => {
+            const newTokenKey = deprecatedTokens[oldTokenKey];
+            const userToken = token[component] as unknown as Record<string, any>;
```
Comment (MadCcc) [inline components/theme/util/genComponentStyleHook.ts:84]: 这行定义可以拿到 mergedComponentToken 上面去。另外叫 `customComponentToken` 比较好

Comment (MadCcc) [reply components/theme/util/genComponentStyleHook.ts:84]: 这里的类型应该也是 `OverrideTokenWithoutDerivative[ComponentName]`

**Review by MadCcc — APPROVED** [review]

**Pull request merged**